### DISCUSSION TITLE
Adding missing residues into the sequence panel.

### DISCUSSION THREAD

**Discussion by bisejdiu** [molstar/molstar#125] [Q&A] [answered]
Dear all, 

I would like to customize the sequences that are displayed in the sequence panel. For example, if I have a sequence like _VYDPETGTWYG_, my goal is to change it (without affecting the rendered structure) to one of the following: 

1. VYDPE----TGTW-YG
2. VYDPE    TGTW YG

Either of these would be okay. Is it possible to do this and if so how would I do it? 
I tried overwriting the ``missing`` property of the constructor method of `PolymerSequenceWrapper`, but that doesn't work. I also tried to change the _msp-sequence-missing_ elements but couldn't get that to work either. I thought of either changing how cif files are loaded by Mol* or changing them before they are loaded, but that also changes the way structures are drawn on the screen and I only want to change the sequence. 

Thank you for any help/comment/suggestion.

**Comment by dsehnal** [comment, marked as answer]
Your approach to modify ``PolymerSequenceWrapper`` is good I think.

But what you need to do is to construct a new ``Sequence`` object as well which would contain ``-`` in the ``code``/``label`` columns. Then you would need to modify all the ``seqId``/``residueLabel``/... to return correct values based on the modifications of the ``sequence`` object that you made.

**Reply by bisejdiu** [reply]
Thank you for your help @dsehnal! I fixed this a few days ago and it's working quite nicely.
